### ISSUE TITLE
Blazor WASM .Net Core Hosted IIS Web Deploy published web.config

### ISSUE THREAD

**Issue by adrianwright109** [dotnet/aspnetcore#30853]
When I publish my Blazor Web Assembly ASP.NET Core Hosted application using Visual Studio and an IIS Web Deploy publish profile.

The following `web.config` file is generated into the IIS Website folder.

```
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <location path="." inheritInChildApplications="false">
    <system.webServer>
      <handlers>
        <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModuleV2" resourceType="Unspecified" />
      </handlers>
      <aspNetCore processPath="dotnet" arguments=".\MyApp.Server.exe" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" hostingModel="inprocess" />
    </system.webServer>
  </location>
</configuration>
<!--ProjectGuid: -->
```

I am curious to know why the `<aspnetCore />` element references the server `.exe` not the server `.dll`

The app seems to be functioning correctly hosted in IIS but it appears that all the online examples of this same generated `web.config` file reference the server `.dll` not the server `.exe` in the `<aspnetCore />` `arguments` attribute.

e.g. `arguments=".\MyApp.Server.exe"` would be `arguments=".\MyApp.Server.dll'`

Just wondering why my published app is using the server `.exe`?

It seems that to run the published app from the command line using Kestrel that the `dotnet` command requires the server `.dll` not the `.exe` as the command argument.

e.g. `dotnet .\MyApp.Server.dll`

Anyone able to clarify what the `web.config` should be using?

Thanks

**Comment by BrennanConroy** [comment]
Do you have a repro, or a web.config already in your project?

@BillHiebert

**Comment by adrianwright109** [comment]
There is no `web.config` in my project.  The only `web.config` files I can find are the ones created by the publishing process.
![image](https://user-images.githubusercontent.com/47532258/111440374-2d043e00-86fe-11eb-88ef-43430e5d9faa.png)

I will try to create a minimal repo and the project I'm working on cannot be shared publicly.

**Comment by adrianwright109** [comment]
I have created a new Blazor Web Assembly Hosted project and published it using VS and repo'd the issue.

The code for this is in this repo: https://github.com/adrianwright109/WasmNoAuth

**N.B.** I have also included the published output in the repo so you can see what it produced.

The `web.config` that it created has processPath as `processPath=".\MB5.Server.exe"` rather than `processPath=".\MB5.Server.dll"` as expected.

**Comment by BillHiebert** [comment]
@vijayrkn

**Comment by vijayrkn** [comment]
It looks like Blazor Server projects are always published with a RID & Self-contained and that is why the web.config is referring to the executable. 

To have the web.config use the dll instead, you can set this property to the pubxml - https://github.com/adrianwright109/WasmNoAuth/blob/master/MB5/Server/Properties/PublishProfiles/FolderProfile.pubxml
`	<_UseAppHost>false</_UseAppHost>`

@pranavkm - Is there a reason Blazor server projects are always set as self-contained?

**Comment by vijayrkn** [comment]
It looks like '`UseAppHost'` might be getting set when ever you are publishing with a RID. If your intend is to publish purely a framework dependent app, then you can remove the RID from the pubxml and no other changes would be needed. 

`    <RuntimeIdentifier>win-x64</RuntimeIdentifier>`

**Comment by mkArtakMSFT** [comment]
This is being fixed in 6.0 preview7.
